src/depthai_wrapper.cpp

```diff
@@ -0,0 +1,169 @@
+{
+
+    if (config_json_.contains("streams"))
+    {
+        for (auto it : config_json_.at("streams"))
+        {
+            // "metaout"
+            if (it.is_string())
+            {
+                if (it.get<std::string>() == "metaout")
+                    continue;
+                stream_names_.push_back(it.get<std::string>());
+                if (it.get<std::string>() == "previewout")
+                {
+                    std::shared_ptr<cv::Mat> img =
+                        std::make_shared<cv::Mat>(rgb_height_, rgb_width_, CV_8UC3);
+                    image_streams_.push_back(img);
+                }
+                else
+                {
+                    std::shared_ptr<cv::Mat> img =
+                        std::make_shared<cv::Mat>(mono_height_, mono_width_, CV_8UC3);
+                    image_streams_.push_back(img);
+                }
+            }
+            else
+            {
+                const auto& name = it.at("name").get<std::string>();
+                // streams.emplace_back(name);
+                // auto &stream = streams.back();
+                // std::cout << name << " -streams" << std::endl;
+                stream_names_.push_back(name);
+                if (name == "previewout")
+                {
+                    std::shared_ptr<cv::Mat> img =
+                        std::make_shared<cv::Mat>(rgb_height_, rgb_width_, CV_8UC3);
+                    image_streams_.push_back(img);
+                }
+                else if (name == "depth")
+                {
+                    std::shared_ptr<cv::Mat> img =
+                        std::make_shared<cv::Mat>(mono_height_, mono_width_, CV_16UC1);
+                    image_streams_.push_back(img);
+                }
+                else if (name == "disparity_color")
+                {
+                    std::shared_ptr<cv::Mat> img =
+                        std::make_shared<cv::Mat>(mono_height_, mono_width_, CV_16UC1);
+                    image_streams_.push_back(img);
+                }
+            }
+        }
+    }
+}
+
+DepthAI::~DepthAI() = default;
+
+// How about creating streams for everything in the vector in constructor and then updating them
+// each time get frames is called ?
+
+// TODO: Crash on size mismatch. DO this later.
+// TODO: save config as json and check if the streams are in the config and crash if they aren't
+
+void DepthAI::get_frames(std::unordered_map<std::string, CV_mat_ptr>& output_streams)
+{
+    // std::cout << "mono it-----" << mono_height_ << " ~~ " << mono_width_ << std::endl;
+
+    int count = stream_names_.size();
+    std::vector<std::string>::iterator it;
+    std::vector<bool> dirty_check(stream_names_.size(), false);
+    while (count)
+    {
+        packets_ = pipeline_->getAvailableNNetAndDataPackets(true);
+
+        for (auto sub_packet : std::get<1>(packets_))
+        {
+            it = std::find(stream_names_.begin(), stream_names_.end(), sub_packet->stream_name);
```

Comment:
Ah this is not a good idea. Why are you performing linear search each time inside a vector? Why not create `stream_names_` an `unordered_set` instead? So that it is O(1) search.

Reply:
so if I do this how will I find the index of the dirty_check ? one way I can think of is making it also a map and updating it. 
what do you think I should do ?